nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/AbstractExecuteSQL.java

```diff
@@ -312,12 +312,6 @@ public void onTrigger(final ProcessContext context, final ProcessSession session
                                 // If we've reached the batch size, send out the flow files
                                 if (outputBatchSize > 0 && resultSetFlowFiles.size() >= outputBatchSize) {
                                     session.transfer(resultSetFlowFiles, REL_SUCCESS);
-                                    // Need to remove the original input file if it exists
-                                    if (fileToProcess != null) {
-                                        session.remove(fileToProcess);
-                                        fileToProcess = null;
-                                    }
-                                    session.commit();
```

Comment:
We can't remove this `session.commit()` because this is required when 'Output Batch Size' is higher than 0. The purpose is committing FlowFiles before iterating through whole result set so that downstream flow can process those gradually.

Also, the `if (fileToProcess != null) {` block is required to address NIFI-6040. This PR doesn't have the issue reported by NIFI-6040 because it also removes the `session.commit()`.

I think the correct solution is removing the only line 318, `fileToProcess = null;`.

Reply:
Hi @ijokarumawak 
Thanks for your reply. It will be failed if we only remove `fileToProcess = null;` The failure occurs in the next iteration in `resultSetFF = session.create(fileToProcess);`.

I can get all attributes of `fileToProcess` in advance, and create flowfile in the iteration with `session.create(attributeMap)` instead of `session.create(fileToProcess);`. But it loses the parent linkage.

Reply:
Hi @avseq1234 That's true. I agree with the approach using `session.create(attributeMap`. Losing the parent linkage is acceptable in this case I believe, since current master doesn't have it, too, and also it's impossible to support partial process session commit (which requires removing the incoming FlowFile) and reporting FORK lineage at later iteration (which requires the already removed FlowFile) at the same time.

We can optionally add the incoming FlowFile's UUID to an additional FlowFile attribute of the generated resultSetFFs such as `input.flowfile.uuid` so that users can search parent FlowFile lineage if needed.

Reply:
Hi @ijokarumawak 
I use the following snippet to create each new flowfile in the iteration. 
` FlowFile resultSetFF = session.create();` 
` resultSetFF = session.putAllAttributes(resultSetFF, inputFileAttrMap);`

I think it is more consistent than using ` fileToProcess == null`  condition to create new flowfile. I also add `input.flowfile.uuid` attribute of generated resultSetFFs.

Reply:
The way that I mentioned will break some test. I am going to fix it later.

Reply:
I modify the code because the way that I mentioned before will break some test.